File: backend/src/race_command_center/models/decision.py
from pydantic import BaseModel, Field


class CrewChiefDecision(BaseModel):
    decision_id: str
    session_id: str | None = None
    recommendation_id: str | None = None

    title: str
    decision_type: str
    risk_level: str

    status: str = "proposed"
    proposed_by: str = "system"
    approved_by: str | None = None

    evidence: list[dict] = Field(default_factory=list)
    simulation_id: str | None = None
    workflow_id: str | None = None

    created_at: str | None = None
    decided_at: str | None = None
    notes: str | None = None


class DecisionApprove(BaseModel):
    approved_by: str = "crew_chief"
    notes: str | None = None


class DecisionReject(BaseModel):
    reason: str
    rejected_by: str = "crew_chief"

File: frontend/src/types/decision.ts
export type DecisionStatus =
  | "draft"
  | "proposed"
  | "simulation_required"
  | "simulation_completed"
  | "crew_chief_review"
  | "approved"
  | "rejected"
  | "applied"
  | "archived";

export type CrewChiefDecision = {
  decision_id: string;
  session_id?: string;
  recommendation_id?: string;
  title: string;
  decision_type: string;
  risk_level: string;
  status: DecisionStatus;
  proposed_by: string;
  approved_by?: string;
  evidence: Record<string, unknown>[];
  simulation_id?: string;
  workflow_id?: string;
  created_at?: string;
  decided_at?: string;
  notes?: string;
};
